src/main/java/frc/robot/Systems/Chassis/Chassis.java

```diff
@@ -27,52 +42,68 @@
 import edu.wpi.first.wpilibj.drive.RobotDriveBase;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
 import frc.robot.Constants.Constants;
 import frc.robot.Constants.RobotConstants;
 
 public class Chassis extends SubsystemBase {
   public SwerveModule[] modules;
-  public ChassisSpeeds chassisSpeed;
-  public SwerveModuleState[] states;
+  private Pigeon2 pigeon;
+
+  public ChassisSpeeds setChassisSpeed;
+  public SwerveModuleState[] setStates;
+
+  public ChassisSpeeds getChassisSpeed;
+  public SwerveModuleState[] getStates;
+  public SwerveModulePosition[] getPositions;
   
   private SwerveDriveKinematics kinematics;
   private SwerveDrivePoseEstimator poseEstimator;
-  private Pigeon2 pigeon;
+  private PhotonCamera ATCamera;
+  private PhotonPoseEstimator photonEstimator;
+  private AprilTagFieldLayout aprilTagFieldLayout;
```

Comment:
I think the pose estimator doesn't actually belong only the Chassis class. It should be passed in and the `update` method should be called periodically with the module locations. But the chassis has nothing to do with April tag detection